src/modules/dashboard/components/ExpenditurePage/Payments/LockedPayments.tsx

```diff
@@ -69,75 +77,91 @@ const LockedPayments = ({
-        return null;
-      }
-
-      const { amount, time } = recipientDelay;
-
-      if (!amount) {
-        return null;
-      }
-
-      if (time === 'months') {
-        return amount === '1' ? `, ${amount}mth` : `, ${amount}mths`;
-      }
-      return `, ${amount}${time.slice(0, 1)}`;
-    },
-    [],
+  const pendingCreations = pendingChanges?.recipients?.filter(
+    (recipient) => recipient?.created,
   );
 
   return (
     <div className={styles.paymentContainer}>
       <div className={styles.recipientContainer}>
         <div className={styles.payments}>
           <FormattedMessage {...MSG.payments} />
+          {activeState?.id !== Stage.Claimed && (
+            <span className={styles.editIcon}>
+              <Icon
+                name="edit"
+                appearance={{ size: 'medium' }}
+                title="Edit expenditure"
+                onClick={editForm}
+              />
+            </span>
+          )}
         </div>
         {recipients?.map(({ claimed, claimDate, ...recipient }, index) => {
           const isOpen =
             expandedRecipients?.find((idx) => idx === index) !== undefined;
-          const recipientName =
-            recipient.recipient.profile.displayName ||
-            recipient.recipient.profile.username ||
-            '';
+
+          const inPendingState = pendingChanges?.recipients?.find(
+            (item) => item.id === recipient.id,
+          );
 
           return (
             <div className={styles.singleRecipient} key={recipient.id}>
+              {colony && (
+                <>
+                  <RecipientHeader
+                    isOpen={isOpen}
+                    onToggleButtonClick={onToggleButtonClick}
+                    inPendingState={!!inPendingState}
+                    recipient={recipient}
+                    index={index}
+                    colony={colony}
+                    ClaimTag={
+                      activeState?.id === Stage.Released &&
+                      claimDate &&
+                      !isCancelled && (
+                        <ClaimTag
+                          claimDate={claimDate}
+                          claimed={claimed}
+                          activeState={activeState}
+                          pendingMotion={pendingMotion}
+                        />
+                      )
+                    }
+                  />
+                  <LockedRecipient
+                    recipient={{
+                      ...recipient,
+                      isExpanded: isOpen,
+                    }}
+                    colony={colony}
+                    pendingChanges={pendingChanges}
+                  />
+                </>
+              )}
+            </div>
+          );
+        })}
+      </div>
+      <div className={styles.recipientContainer}>
+        {pendingCreations?.map((newRecipient, index) => {
+          return (
+            <div className={styles.singleRecipient} key={newRecipient.id}>
```

Comment:
Same here, please use the count as well as the recipient.id i.e. address as the key. For example `0xaEsd0...wE73_1`